src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -324,38 +324,28 @@ else if (currentAst.getType() != TokenTypes.VARIABLE_DEF) {
      * Returns the distance to variable usage for in the child node.
      *
      * @param childNode child node.
-     * @param varIdent variable variable identifier.
      * @param currentDistToVarUsage current distance to the variable usage.
      * @return the distance to variable usage for in the child node.
      */
-    private static int getDistToVariableUsageInChildNode(DetailAST childNode, DetailAST varIdent,
+    private static int getDistToVariableUsageInChildNode(DetailAST childNode,
                                                          int currentDistToVarUsage) {
-        DetailAST examineNode = childNode;
-        if (examineNode.getType() == TokenTypes.LABELED_STAT) {
-            examineNode = examineNode.getFirstChild().getNextSibling();
```

Comment:
Can you explain why this issue impacts labels specifically? Nothing in the issue mentioned this type.

Reply:
@rnveach Yes issue has nothing to do with labels but this condition was a surviving mutation reported by Pitest after my changes. it is just a redundant condition now and is being handled by the default case in the switch. we don't need to get the sibling of its child (loop statement) because the distance will be added by 1 but if we remove this condition completely the label will hit default case in the switch and the distance will be added by 1 also 
So it is safe to remove

Reply:
Can you generate a custom diff report? Make `allowedDistance` = 1, before image (base) being the code in this PR as it is now, the after image (patch) being the code in this PR with this label code restored. If no differences are found, then I am good.

Reply:
> it is just a redundant condition now and is being handled by the default case in the switch

The code in the default switch has nothing in common with the label code. One gets the first child and next sibling, the other calculates distance value.

> was a surviving mutation reported by Pitest after my changes

Pitest just reports that there are no tests to prove the reported code is necessary or not, not the worth of keeping the production code. It is up to us to determine by examination if pitest is reporting an actual issue with the production code, or if simply more tests are needed.

Reply:
> The code in the default switch has nothing in common

I didn't mean common in code. I mean that after we finish the switch block, we will achieve the same result whether we remove the condition or not

- if we leave the condition we will [get the first child and next sibling](https://github.com/checkstyle/checkstyle/blob/ccde012394294f31d1cc28c51e03c6a622424663/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java#L335) which is `LITERAL_WHILE` or `LATERAL_FOR` then we will hit [this line](https://github.com/checkstyle/checkstyle/blob/9ab3556eec80d5b2c3a7430d518a6ac6a93dcf72/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java#L345) where we increment the distance by 1
- if we remove the condition we will hit [this line in the default case](https://github.com/checkstyle/checkstyle/blob/ccde012394294f31d1cc28c51e03c6a622424663/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java#L359) because `LABELED_STAT.findFirstToken(TokenType.SLIST) == null` so we increment the distance by 1 also 

this was not the case before my changes. the condition was necessary because when we  [get the first child and next sibling](https://github.com/checkstyle/checkstyle/blob/ccde012394294f31d1cc28c51e03c6a622424663/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java#L335) we enter the switch case and set distance to zero this can't be done in the default 
but now we want to add 1 not set to zero therefore this condition is redundant now

Reply:
@rnveach please review if this explanation is good if not I can generate report as you requested above

Reply:
@rnveach please review if this explanation is enough if not I can generate report as you requested above

Reply:
Please try generating the report.

Reply:
@rnveach I tried so many times and got these errors while generating the report can you please tell me how to solve it? I used to use GitHub actions I haven't used manual before
this is the command that I ran ` D:\contribution\checkstyle-tester> groovy diff.groovy --localGitRepo  D:\checkstyle --baseBranch issue-3502 --patchBranch issue-3502-revertlabel --config my_check.xml --listOfProjects projects-to-test-on.properties`

```
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:28 min
[INFO] Finished at: 2024-02-28T02:39:20+02:00
Testing Checkstyle started
   [delete] Deleting directory D:\contribution\checkstyle-tester\src\main\java
Cloning git repository 'checkstyle' to repositories\checkstyle folder ...
Cloning into 'repositories\checkstyle'...
error: RPC failed; curl 92 HTTP/2 stream 0 was not closed cleanly: CANCEL (err 8)
error: 3941 bytes of body are still expected
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
error: RPC failed; curl 92 HTTP/2 stream 0 was not closed cleanly: CANCEL (err 8)
error: 4652 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Cloning into 'repositories\checkstyle'...
error: 5107 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Cloning into 'repositories\checkstyle'...
error: 5561 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Cloning into 'repositories\checkstyle'...
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at diff.generateCheckstyleReport(diff.groovy:271)
        at diff$generateCheckstyleReport$11.callCurrent(Unknown Source)
        at diff.launchCheckstyleReport(diff.groovy:232)
        at diff$launchCheckstyleReport$7.callCurrent(Unknown Source)
        at diff.run(diff.groovy:31)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

Reply:
@mahfouz72 Restore the code and add a suppression to the pitest file in this PR. I talked to another admin and we add a suppression for now.

Afterwards, can you report your issue to https://github.com/checkstyle/contribution/issues/ .

Reply:
I am good to extend suppression list in a case like this, we can do a follow up to kill this mutation to make life easier for both @mahfouz72 and reviewers, and get the fix to users.

Reply:
@rnveach @nrmancuso I extended suppression and reported[ issue](https://github.com/checkstyle/contribution/issues/839) for this. is it an individual issue? also, I am wondering if I can use GitHub actions in case that the base isn't  master branch like in this case

Reply:
I don't use GH actions, so I can't say for sure, but I am pretty sure GH actions only works with master and the code in the PR. My ask was you use a separate branch of code, so to use that with GH actions you would need another PR or to crash this PR's state for the run.